### PR TITLE
Add foldername.md pattern to getExampleFilename when component is ind…

### PR DESCRIPTION
**Pull request by DavidHenri008**
Fix issue [#1169](https://github.com/styleguidist/react-styleguidist/issues/1169)
Update to getExampleFilename in order to detect example file when the component definition file is `index.js` and the examples are in FolderName.md.
Test has been added.
Unfortunately all other tests do not pass but it looks like it was like this when I forked.

### CODE REVIEW

File: scripts/schemas/config.js [hunk 1]
```diff
@@ -82,16 +82,17 @@ module.exports = {
 		default: componentPath => {
 			const files = [
 				path.join(path.dirname(componentPath), 'Readme.md'),
-				componentPath.replace(path.extname(componentPath), '.md'),
+				// ComponentName.md
+				path.join(componentPath.replace(path.extname(componentPath), '.md')),
```
Comment (sapegin) [inline scripts/schemas/config.js]: Looks like you don't need `path.join` here.

Comment (DavidHenri008) [reply scripts/schemas/config.js]: I will remove it.

**Comment by DavidHenri008** [comment]
@sapegin This is my very first PR and I am working on Windows. I have a hard time to complete the PR for two reasons:
1. When I run `npm run test` I am getting many errors that has nothing to do with my modifications. I cloned the master branch in a different repository and I still get the same errors. Do you know if tests (`npm run test`) are successful?
2. Each time I run `npm run test` or `npm start` I am getting 400+ changes in git. But there is no apparent differences. I suspect a crlf/lf issue. But I do not know how to solve this. I am using VSCode as IDE. I think this is the cause of the final differences (since my last commit) that you see in my package-lock.json.

Thanks.

**Comment by DavidHenri008** [comment]
As a note I found out that many tests based on snapshots returns error because of path delimiter, ex:
-       "filepath": "components/Price/Price.js",
+       "filepath": "components\\Price\\Price.js", 
Again I just do not know how to fix this on my VSCode Windows setup.

**Comment by sapegin** [comment]
1. Tests are passing in master — you can see it in the Readme — [Build Status](https://travis-ci.org/styleguidist/react-styleguidist)

    But I'm sure we have some issues on Windows ;-( If you're feeling adventurous and want to help ;-) it would be super cool.

2. It could be because your Git is converting line endings on checkout — https://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/

**Comment by sapegin** [comment]
> Again I just do not know how to fix this on my VSCode Windows setup.

The issue with tests is in the repo, not in your setup. But I don't know what could be the right solution here, so any help is appreciated.

**Comment by sapegin** [comment]
I'll update the snapshot, thanks for the work! ;-)

I think removing paths from snapshots is the right way. We can use something like [slash](https://github.com/sindresorhus/slash) to compare paths outside snapshots.

**Pull request merged**